File: telegram-bot-core/src/main/java/ru/tggc/telegrambotcore/registry/resolver/UpdateResolver.java
package ru.tggc.telegrambotcore.registry.resolver;

import com.pengrad.telegrambot.model.Update;
import org.springframework.stereotype.Component;

import java.lang.reflect.Parameter;

@Component
public class UpdateResolver implements ParameterResolver<Update> {

    @Override
    public boolean supports(Parameter p) {
        return p.getType().equals(Update.class);
    }

    @Override
    public Update resolve(Parameter parameter, HandlerCtx ctx) {
        return ctx.update();
    }
}

File: telegram-bot-core/src/main/kotlin/ru/tggc/telegrambotcore/registry/resolver/UpdateResolver.kt
package ru.tggc.telegrambotcore.registry.resolver

import com.pengrad.telegrambot.model.Update
import org.springframework.stereotype.Component
import java.lang.reflect.Parameter

@Component
class UpdateResolver : ParameterResolver<Update?> {
    override fun supports(p: Parameter?): Boolean = p?.getType() == Update::class.java

    override fun resolve(parameter: Parameter?, ctx: HandlerCtx): Update? = ctx.update
}
